### PR TITLE
Report video decode error

### PR DESCRIPTION
**Pull request by mikepantazoglou**

### CODE REVIEW

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp [hunk 1]
```diff
@@ -1943,7 +1943,8 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
             // Let the mediaPlayerClient handle the stream error, in this case the HTMLMediaElement will emit a stalled event.
             GST_ERROR("Decode error, let the Media element emit a stalled event.");
             m_loadingStalled = true;
-            break;
+            error = MediaPlayer::NetworkState::DecodeError;
```
Comment (eocanha) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:1946]: <strike>If you're triggering a DecodeError here, the code block is equivalent to the one in the `else if` block. Please, merge both blocks into a single one, triggered by an OR (`||`) condition.</strike>

(Please forget the comment above, I hadn't realized about the `m_loadingStalled` line, which makes this code block different)

**Comment by eocanha** [comment]
Please, could you an an explanation in the commit log about why the current code is causing problems and why the change is justified? Thanks!

**Review by eocanha — CHANGES_REQUESTED** [review]

**Comment by mikepantazoglou** [comment]
Without the proposed change, the player does not set the error to MediaPlayer::NetworkState::DecodeError when a GST_STREAM_ERROR or a GST_STREAM_ERROR_TYPE_NOT_FOUND occurs, but expects from the application to do so if needed via the stalled event. However the latter is too generic for the application to determine the actual cause of error and decide whether this is a decode error or not.

We have seen this problem in one of the use cases we are working on, where the application attempts to play invalid video content and its code translates decode errors to an internal 'fatal' state. If the application relied just on the stalled event, it would not be possible to differentiate from that event if there has been a decode error or if the event was emitted due to some other deficiency (e.g. temporary bandwidth issues).

**Comment by vivienne-w** [comment]
This PR is tracked upstream as: https://bugs.webkit.org/show_bug.cgi?id=257195 and https://github.com/WebKit/WebKit/pull/14237.

It landed upstream as https://github.com/WebKit/WebKit/commit/47683cbce224b66ec4be118dc9279b730d204e8a, in wpe-2.28 as https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/a8cb57195b377b58b8f5ef13d8c60f9b36ed2c7e and in wpe-2.38 as https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/704d9a8c2d1f72e8f475308a7364a5e34ff4dd0d so I'm closing this PR.

**Pull request closed without merging**